### ISSUE TITLE
User Sync creating duplicate adobeID/federatedID accounts

### ISSUE THREAD

**Issue by mac-reid** [adobe-apiplatform/user-sync.py#559]
We recently noticed we've been getting duplicate federatedID accounts for folks who already have an adobeID account. I see a few issues on this repo that indicate that this is not the expected behavior. The example below is for my account only. Duplicates are created for some (not all) of our users that have an adobeID account.

We are running user-sync 2.5.0 on python 2.7.5/RHEL 7 as the following:
```
/opt/user-sync/user-sync -c /opt/user-sync/cc/user-sync-config.yml --users mapped --process-groups --adobe-only-user-action exclude
```
Our conf file is:
```
---
adobe_users:
  connectors:
    umapi: "/opt/user-sync/cc/connector-umapi.yml"
directory_users:
  user_identity_type: federatedID
  default_country_code: US
  connectors:
    ldap: "/opt/user-sync/connector-ldap.yml"
  groups:
  - directory_group: Group Name
    adobe_groups:
    - Group Name
limits:
  max_adobe_only_users: 100%
logging:
  console_log_level: debug
```

The invocation options:

```
2020-01-21 14:44:33 7781 DEBUG main - -------- Resulting invocation options --------
2020-01-21 14:44:33 7781 DEBUG main -   update_user_info: False
2020-01-21 14:44:33 7781 DEBUG main -   test_mode: False
2020-01-21 14:44:33 7781 DEBUG main -   users: ['mapped']
2020-01-21 14:44:33 7781 DEBUG main -   adobe_only_user_list: None
2020-01-21 14:44:33 7781 DEBUG main -   exclude_strays: True
2020-01-21 14:44:33 7781 DEBUG main -   encoding_name: utf8
2020-01-21 14:44:33 7781 DEBUG main -   adobe_only_user_action: ['exclude']
2020-01-21 14:44:33 7781 DEBUG main -   adobe_group_mapped: False
2020-01-21 14:44:33 7781 DEBUG main -   strategy: sync
2020-01-21 14:44:33 7781 DEBUG main -   connector: ['ldap']
2020-01-21 14:44:33 7781 DEBUG main -   process_groups: True
2020-01-21 14:44:33 7781 DEBUG main -   config_filename: /opt/user-sync/cc/user-sync-config.yml
2020-01-21 14:44:33 7781 DEBUG main -   directory_connector_type: ldap
2020-01-21 14:44:33 7781 DEBUG main -   user_filter: None
2020-01-21 14:44:33 7781 DEBUG main -   directory_group_mapped: True
2020-01-21 14:44:33 7781 DEBUG main -   stray_list_output_path: None
2020-01-21 14:44:33 7781 DEBUG main -   adobe_users: ['all']
2020-01-21 14:44:33 7781 INFO main - -------------------------------------
```

The relevant output regarding my user account:
```
2020-01-21 14:46:29 7781 DEBUG processor - Excluding Adobe-only user: adobeID,mareid@mtu.edu,
2020-01-21 14:46:29 7781 INFO processor - Creating user with user key: federatedID,mareid@mtu.edu,
2020-01-21 14:46:29 7781 DEBUG umapi.action - Added action: {"do": [{"createFederatedID": {"lastname": "Reid", "option": "ignoreIfAlreadyExists", "email": "mareid@mtu.edu", "firstname": "Mac", "country": "US"}}, {"add": {"product": ["Group Name"]}}], "requestID": "action_5", "user": "mareid@mtu.edu"}
```

However, in the adobe console:

![duplicate-adobe-accounts](https://user-images.githubusercontent.com/5366758/72838743-e385a380-3c5e-11ea-805b-0fdfef6c2582.png)

I'm not sure if this is a configuration issue on our end or what. We'd really like to not have these duplicate accounts created though.

**Comment by vossen-adobe** [comment]
Did you by any chance to a migration from adobe ID to federated using CSV?  This will create duplicates for admins

**Comment by mac-reid** [comment]
No, I don't believe we ever used csv and this is occurring with non-admin accounts. I can delete my federatedID account, run the user-sync script, and I will have a new federatedID account when the script finishes. I'm curious how a new account is created if the script isn't creating it.

**Comment by adorton-adobe** [comment]
The User Management API has had a long-standing issue where it doesn't allow a federated ID to be created when an Adobe ID with the same email address already exists. It is possible to do this manually in the admin console, or using other tools like the bulk CSV tools. The UMAPI however will report a "success" status when creating such a Federated user, but no account will be created.

I'm wondering if this issue has been resolved. I'll do some investigating.

**Comment by mac-reid** [comment]
Have you had any chance to review this? We're still seeing this issue (just tested again today).

**Comment by adorton-adobe** [comment]
I can't reproduce this behavior. I'm still getting the "false positive" behavior I described previously. My guess is the duplicate accounts were created as a result of an identity migration or a CSV upload, since the console UI nor the UMAPI will allow you to create duplicate accounts in this manner. The best the UST can do is skip the creation of a Federated ID if an Adobe ID with the same email address already exists. We've implemented this behavior as of the latest release - v2.6.1.
